src/esc/environment_fs_provider.ts

```diff
@@ -141,8 +155,17 @@ export class EnvironmentFileSystemProvider implements vscode.FileSystemProvider,
         await this.api.patchEnvironment(org, project, envName, contentStr);
         const uriStr = uri.toString();
         this.updates.set(uriStr, Date.now());
-        this.sizes.set(uriStr, contentStr.length);
-        this._emitter.fire([{ type: vscode.FileChangeType.Changed, uri }]);
+        
+        this.sizes.delete(uriStr);
+        const watchFires = new Map<vscode.Uri, vscode.FileChangeEvent>();
+        watchFires.set(uri, { type: vscode.FileChangeType.Changed, uri });
```

Comment:
Please explain better what this is doing.  Seems like there might be a watch for the env definition and for the "open" environment, and a change to the definition causes a change to both.

Reply:
Yes, that is exactly right. This is firing regardless of watch for the URI that changes, and look at the watches for any `/open/*` of the same environment. So if you have the open in preview on the right it will auto refresh.